axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/AccountingCutOffServiceImpl.java

```diff
@@ -362,13 +366,20 @@ public Move generateCutOffMove(
 
   protected boolean checkStockMoveLine(
       StockMoveLine stockMoveLine, Product product, boolean includeNotStockManagedProduct) {
-    if (stockMoveLine.getRealQty().compareTo(BigDecimal.ZERO) == 0
-        || product == null
-        || (!includeNotStockManagedProduct && !product.getStockManaged())) {
+    Beans.get(StockMoveLineServiceSupplychain.class).setInvoiceStatus(stockMoveLine);
+    System.err.println(stockMoveLine.getAvailableStatusSelect());
+    if (stockMoveLine.getAvailableStatusSelect() == 3
```

Comment:
Remove nested if-else and directly add condition using qty instead of availableStatusSelect